pynumaflow/function/async_server.py

```diff
@@ -138,21 +138,34 @@ async def MapFn(
         """
         # proto repeated field(keys) is of type google._upb._message.RepeatedScalarContainer
         # we need to explicitly convert it to list
-        res = await self.__invoke_map(
-            list(request.keys),
-            Datum(
-                keys=list(request.keys),
-                value=request.value,
-                event_time=request.event_time.event_time.ToDatetime(),
-                watermark=request.watermark.watermark.ToDatetime(),
-                metadata=DatumMetadata(
-                    msg_id=request.metadata.id,
-                    num_delivered=request.metadata.num_delivered,
+        try:
+            res = await self.__invoke_map(
+                list(request.keys),
+                Datum(
+                    keys=list(request.keys),
+                    value=request.value,
+                    event_time=request.event_time.event_time.ToDatetime(),
+                    watermark=request.watermark.watermark.ToDatetime(),
+                    metadata=DatumMetadata(
+                        msg_id=request.metadata.id,
+                        num_delivered=request.metadata.num_delivered,
+                    ),
                 ),
-            ),
-        )
+            )
+        except Exception as e:
+            context.set_code(grpc.StatusCode.UNKNOWN)
```

Comment:
Instead of using `UNKNOWN` should we use `INTERNAL` as this is application layer error?

Reply:
i followed the guidance here - https://grpc.github.io/grpc/core/md_doc_statuscodes.html